File: functions/src/data/models/user-full-model.ts
import {UserFull} from "../../domain/entities/user-full";
import {UserPublicModel} from "./user-public-model";


export class UserFullModel extends UserFull {
    static readonly kFcmToken = "fcmToken";
    constructor(
        uid: string,
        firstName: string,
        lastName: string,
        agoraUid: number,
        fcmToken: string,
    ) {
        super(uid, firstName, lastName, agoraUid, fcmToken);
    }

    static fromData(data:any) : UserFullModel | undefined {
        if (!data) {
            return undefined;
        }
        return new UserFullModel(
            data[UserPublicModel.kUid],
            data[UserPublicModel.kFirstName],
            data[UserPublicModel.kLastName],
            data[UserPublicModel.kAgoraUid],
            data[UserFullModel.kFcmToken],
        )
    }

}

File: flutter_app/lib/core/data/models/user_full_model.dart
import 'package:flutter_group_chat_app_with_firebase/core/data/models/user_public_model.dart';
import 'package:flutter_group_chat_app_with_firebase/core/domain/entities/user_full.dart';


class UserFullModel extends UserFull {
  /// Field names:
  static const String kUid = "uid";
  static const String kFcmToken = "fcmToken";

  UserFullModel({required String uid, required String firstName, required String lastName, required String? fcmToken})
      : super(uid: uid, firstName: firstName, lastName: lastName, fcmToken: fcmToken,);

  static UserFullModel? fromMap({required Map<String,dynamic>? userFull}) {
    if (userFull == null) {
      return null;
    }
    return UserFullModel(
      uid: userFull[kUid],
      fcmToken: userFull[kFcmToken],
      firstName: userFull[UserPublicModel.kFirstName],
      lastName: userFull[UserPublicModel.kLastName],
    );
  }

  Map<String, dynamic> toMap() => {
    kUid: uid,
    kFcmToken: fcmToken,
    UserPublicModel.kFirstName: firstName,
    UserPublicModel.kLastName: lastName,
  };

}